### PR TITLE
Shooter: Fix for #3, Create a new shooter subsystem

### LINKED ISSUE #3: Shooter Subsystem
The shooter subsytem must be created, this involves making new substem code, new methods and commands to conrol it. As well as new configurations, new wiring, etc, this is a multi-member project.

### PR DESCRIPTION
**Pull request by KenwoodFox**
Create a shooter subsystem

### CODE REVIEW

File: rio/subsystems/yoke.py [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+
+    def kickyGoSloppy(self, degs, curDegs): #raises or lowers the kicky the inputted amount of degrees
+        #degs is the amount to move, curDegs is the degrees off the ground it already is
+        if(curDegs+curDegs<=60 and curDegs+curDegs>=0): #stops the kicky from going underground or into itself
+            self.kicky.set(degs) #moves the kicky
+            return curDegs+degs  #returns the updated degrees
+        else:
+            if(degs>0):
+                self.kicky.set(60-curDegs) #moves the kicky
+                return 60 #returns maximum
```
[the remark this answers is not in the document]
Comment (KenwoodFox) [reply rio/subsystems/yoke.py]: This is a mess

**Review by KenwoodFox — COMMENTED** [review]
This does not work at all.

File: rio/subsystems/yoke.py [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+    def setSpeed(self, speed):
+        """
+        Method to drive, setting
+        a value from 0 to 1 by hand, no speed
+        control required.
+        """
+
+        # Send
+        self.starShooter.set(ControlMode.PercentOutput, speed)
+        self.portShooter.set(ControlMode.PercentOutput, -speed)
```
Comment (KhanSimeoni) [inline rio/subsystems/yoke.py]: Should we be setting the shooter speed to a flat 0 or 1? we probably need a more variable speed for launch angles

**Review by KhanSimeoni — APPROVED** [review]

**Review by KhanSimeoni — CHANGES_REQUESTED** [review]
Oops wrong button I meant to request  changes

File: rio/subsystems/yoke.py (hunk 2 above)
Comment (KenwoodFox) [reply rio/subsystems/yoke.py]: True but only sloppyshooter calls set speed.

We should idealy have a `setVelocity()` function that can take a value in m/s and set the PID on the shooters

File: rio/subsystems/yoke.py [hunk 3]
```diff
@@ -93,18 +93,15 @@ def setAngle(
...
+    ):  # raises or lowers the shooter the inputted amount of degrees
+        # rot2d is what to set it to
+
+        curRads = rot2d.radians()  # converts rottion 2d to radians
+
+        currentRef = curRads / (2 * math.pi)  # (radians) converted to rotations
+
+        self.kickerPID.setReference(
+            currentRef, CANSparkMaxLowLevel.ControlType.kPosition
+        )  # updating the pid target
```
Comment (KenwoodFox) [inline rio/subsystems/yoke.py]: Kicker dosent need pid, its just a kicker

**Pull request closed without merging**